src/helpers/files.js

```diff
@@ -149,6 +150,21 @@ const FilesWorkspacePlugin = {
 			render: this.render.bind(this),
 			priority: 10,
 		})
+
+		fileList.filesClient.addFileInfoParser((_response, data) => {
+			const dir = (data.mimetype === 'httpd/unix-directory')
+				? data.path + (data.path.endsWith('/') ? '' : '/') + data.name
+				: data.path
+			if (dir !== fileList.getCurrentDirectory()) {
+				return
+			}
+			if (data.mimetype === 'httpd/unix-directory') {
+				this.vm.folder = { ...data }
+			}
+			if (data.mimetype === 'text/markdown' && data.name === 'Readme.md') {
```

Comment:
This should also handle other filenames that are taken into consideration:

https://github.com/nextcloud/text/blob/adf5a760a04b50ec4e538f7b2c9a7d8549ef4cae/lib/Service/WorkspaceService.php#L30-L46

A different approach might also be to extend the DAV plugin in order to return the filename of the rich workspace file as a dedicated property, similar to https://github.com/nextcloud/text/blob/master/lib/DAV/WorkspacePlugin.php#L99-L114 which would already return the content.

Reply:
The second approach would have the benefit that we only maintain one codepath on how the workspace file is determined.

Reply:
Oh wow... that looks good. I think we could actually make use of the content.